inference-engine/src/transformations/src/transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,127 @@
+        auto data_shape = std::make_shared<ngraph::opset6::ShapeOf>(decoder_seq_len->input_value(0));
+        auto axisT = ngraph::opset6::Constant::create(seq_len_type, Shape{}, {0});
+        auto indexT = ngraph::opset6::Constant::create(seq_len_type, Shape{1}, {1});
+        auto T = std::make_shared<ngraph::opset6::Gather>(data_shape, indexT, axisT);
+
+        auto axisN = ngraph::opset6::Constant::create(seq_len_type, Shape{}, {0});
+        auto indexN = ngraph::opset6::Constant::create(seq_len_type, Shape{1}, {0});
+        auto N = std::make_shared<ngraph::opset6::Gather>(data_shape, indexN, axisN);
+
+        auto start = opset6::Constant::create(seq_len_type, Shape{}, {1});
+        auto step = opset6::Constant::create(seq_len_type, Shape{}, {1});
+        auto plus1 = opset6::Constant::create(element::i64, Shape{1}, {1});
+        auto plusT = std::make_shared<ngraph::opset6::Add>(T, plus1);
+        auto const_plusT = ngraph::opset6::Constant::create(seq_len_type, Shape{1}, {0});
+        auto plusT_scalar = std::make_shared<ngraph::opset6::Squeeze>(plusT, const_plusT);
+        auto range1T = std::make_shared<ngraph::opset6::Range>(start, plusT_scalar, step, seq_len_type);
+
+        auto mask_shape = std::make_shared<ngraph::opset6::Concat>(
+                OutputVector{T->output(0), N->output(0)}, 0);
+
+        // Generate 2D tensor [T, N] for seq mask
+        auto upper_bounds = std::make_shared<ngraph::opset6::Broadcast>(
+                decoder_seq_len->input_value(1), mask_shape->output(0));
+        auto transpose_upper_bounds = std::make_shared<ngraph::opset6::Transpose>(upper_bounds->output(0),
+                                                                                  ngraph::opset6::Constant::create(seq_len_type,
+                                                                                                                   Shape({2}), {1, 0}));
+        // Compute boolean sequence mask
+        auto bool_seq_mask = std::make_shared<ngraph::opset6::GreaterEqual>(transpose_upper_bounds->output(0),
+                                                                            range1T->output(0));
+
+        // Generate resulted seq mask
+        auto mask_val_true = ngraph::opset6::Constant::create(seq_len_type, Shape{}, {1});
+        auto mask_val_false = ngraph::opset6::Constant::create(seq_len_type, Shape{}, {0});
+        auto seq_mask = std::make_shared<ngraph::opset6::Select>(bool_seq_mask, mask_val_true, mask_val_false);
+        auto transpose_seq_mask = std::make_shared<ngraph::opset6::Transpose>(seq_mask->output(0),
+                                                                              ngraph::opset6::Constant::create(seq_len_type,
+                                                                                                               Shape({2}), {1, 0}));
+        auto transpose_seq_mask_f32 = std::make_shared<ngraph::opset6::Convert>(transpose_seq_mask->output(0), element::f32);
```

Comment:
please explain why we use fp32 type? Is this Convert operation included only in constant path?

Reply:
According to spec of CTCGreedyDecoder (https://docs.openvinotoolkit.org/latest/openvino_docs_ops_sequence_CTCGreedyDecoder_1.html) seq_mask input have floating point type.

Reply:
So that's the question, should it be FP32 or FP16?